frontend/js/heatmap.js

```diff
@@ -0,0 +1,145 @@
+Heatmap = {};
+
+Heatmap.draw = (data, elementID, relativeHippie, size = 500, marginUpdates = {}) => {
+    // set the dimensions and margins of the graph
+    margin = {top: 0, bottom: 0, left: 0, right: 0};
+    Object.assign(margin, marginUpdates);
+
+    let width = size - margin.left - margin.right;
+    let height = size - margin.top - margin.bottom;
+    elementID = elementID || "#protein_heatmap";
+
+// Makes transactions smooth
+    let timeoutFunction = null;
+
+// append the svg object to the body of the page
+    let svg = d3.select(elementID)
+// position heatmap and tooltip
+        .style("display", 'flex')
+        .style("flex-wrap", 'wrap')
+// append svg
+        .append("svg")
+        .attr("width", width + margin.left + margin.right)
+        .attr("height", height + margin.top + margin.bottom)
+        .append("g")
+        .attr("transform",
+            "translate(" + margin.left + "," + margin.top + ")");
+
+// Labels of row and columns
+    let indexes = new Set([
```

Comment:
I want to change this as currently the sorting would favor an ordering based primarily on protein identifiers, rather than experiments. Will implement the change and commit here